package/src/cpumon.cpp

```diff
@@ -106,12 +110,12 @@ void const cpumon::get_hardware_info(nlohmann::json& hw_json) {
     value = std::regex_replace(value, std::regex("^\\s+|\\s+$"), "");
 
     // Fill the JSON with the information
-    for (const auto& metric : metrics) {
-      if (key.compare(metric) != 0) continue;
+    if (key == "Model name" || key == "CPU(s)" || key == "Socket(s)" ||
```

Comment:
Oh no... I actually much preferred the previous version. Now you have these key names encoded twice, *in different places*, which is bad.

I actually doubt that it's any more efficient than the previous `for`loop (which can be slightly improved by adding a `break` after you process the match, because each key can only match once).

Reply:
No problem, let me know how it looks now 👍

Reply:
We re-wrote this bit, so I'm resolving.